homework/index.js

```diff
@@ -21,27 +21,117 @@
+    createAndAppend('td', forks, { html: 'Forks:', class: 'label' });
+    createAndAppend('td', forks, { html: repository.forks });
+    const update = createAndAppend('tr', tableBody);
+    createAndAppend('td', update, { html: 'Updated:', class: 'label' });
+    createAndAppend('td', update, { html: repository.updated_at });
+  }
+
+  function showContributors(rightDiv, repository) {
+    rightDiv.innerHTML = '';
+    createAndAppend('p', rightDiv, {
+      html: 'Contributions',
+      class: 'contributor-head',
+    });
+    fetchJSON(repository.contributors_url, (err, contributors) => {
+      if (err) {
+        createAndAppend('div', rightDiv, {
+          html: err.message,
+          class: 'alert-error',
+        });
+      }
+
+      contributors.map(contributor => {
+        const contributorUL = createAndAppend('ul', rightDiv, {
+          class: 'contributor-ul',
+        });
+        const contributorLi = createAndAppend('li', contributorUL, {
+          class: 'contributor-li',
+        });
+
+        const contributorLink = createAndAppend('a', contributorLi, {
+          target: '_blank',
+          href: contributor.html_url,
+        });
+        createAndAppend('img', contributorLink, {
+          class: 'contributor-image',
+          src: contributor.avatar_url,
+        });
+        const contributorDetails = createAndAppend('div', contributorLink, {
+          class: 'contributor-details',
+        });
+
+        createAndAppend('div', contributorDetails, {
+          html: contributor.login,
+          class: 'contributor-names',
+        });
+        createAndAppend('div', contributorDetails, {
+          html: contributor.contributions,
+          class: 'contributors',
+        });
+      });
+    });
+  }
   function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, { class: 'header' });
+    createAndAppend('p', header, { html: 'HYF Repositories', class: 'hyf' });
+    const select = createAndAppend('select', header, {
+      class: 'select',
+      id: 'list',
+    });
+    const container = createAndAppend('div', root, { class: 'container' });
+    const left = createAndAppend('div', container, { class: 'left_div' });
+    const right = createAndAppend('div', container, { class: 'right_div' });
     fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
       if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        createAndAppend('div', root, {
+          html: err.message,
+          class: 'alert-error',
+        });
       }
+
+      data.sort((a, b) => a.name.localeCompare(b.name));
+      data.map((val, index) => createAndAppend('option', select, { value: index, html: val.name }));
+      document.getElementById('list').addEventListener('change', event => {
```

Comment:
There is no need to use `getElementById` to get reference to the `select` element. You already have a reference in the variable `select` that you created in line 109. So you could simply write:

```js
select.addEventListener('change', event => {
```